### PR TITLE
Memory profile

### LINKED ISSUE #88: Measure memory usage during analysis at cc.inf.elte.hu
Peak is important, average is good to have.  It would be great to measure it per process (e.g.: not the sum of all running clang processes).

### PR DESCRIPTION
**Pull request by dkrupp**
fixes #88

### CODE REVIEW

**Comment by dkrupp** [comment]
if you run xtu-analyzer.py ----record-memory-profile parameter, a new directory will be created .xtu-out/memprof and valgrind massif memory profile output will be stored there for each clang process. These can later be visualized using the ms_print command.

File: tools/xtu-build-new/xtu-analyze.py [hunk 1]
```diff
@@ -365,6 +381,9 @@ def analyze_work():
 
 try:
     os.makedirs(os.path.abspath(mainargs.xtuoutdir))
+    memprofp=os.path.join(os.path.abspath(mainargs.xtuoutdir),"memprof")
```
Comment (Xazax-hun) [inline tools/xtu-build-new/xtu-analyze.py]: Add spaces around `=`, and space after the comma in the argument list.

File: tools/xtu-build-new/xtu-analyze.py [hunk 2]
```diff
@@ -254,6 +264,12 @@ def analyze(directory, command):
             os.path.join(gcov_tmppath, tu_name)
     analyze_cmd = os.path.join(analyze_path, 'analyze-cc') + \
         ' ' + string.join(args, ' ')
+    if mainargs.record_memprof:
+        cmdenv['MEMPROF_PATH'] = memprof_path
+        cmdenv['ANALYZE_BUILD_CLANG_ORIG'] = analyzer_env['ANALYZE_BUILD_CLANG']
+        cmdenv['ANALYZE_BUILD_CLANG'] = memprof_command
+        cmdenv['VALGRIND_PATH'] = os.path.join(valgrind_path,"valgrind")
```
Comment (Xazax-hun) [inline tools/xtu-build-new/xtu-analyze.py]: Space after the comma.

File: tools/xtu-build-new/xtu-analyze.py (hunk 1 above)
Comment (Xazax-hun) [inline tools/xtu-build-new/xtu-analyze.py]: Maybe a full name like `memprof_path` would be more descriptive.

**Pull request closed without merging**